public/app/features/dashboard-scene/sharing/public-dashboards/CreatePublicDashboard.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { useAsync } from 'react-use';
+
+import { DataSourceWithBackend } from '@grafana/runtime';
+import {
+  SceneComponentProps,
+  SceneDataTransformer,
+  SceneGridItem,
+  SceneGridItemLike,
+  SceneGridLayout,
+  SceneGridRow,
+  SceneQueryRunner,
+  VizPanel,
+} from '@grafana/scenes';
+import { CreatePublicDashboard as CreatePublicDashboardComponent } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/CreatePublicDashboard/CreatePublicDashboard';
+import { supportedDatasources } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SupportedPubdashDatasources';
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+
+import { DashboardScene } from '../../scene/DashboardScene';
+import { LibraryVizPanel } from '../../scene/LibraryVizPanel';
+import { PanelRepeaterGridItem } from '../../scene/PanelRepeaterGridItem';
+
+import { SharePublicDashboardTab } from './SharePublicDashboardTab';
+
+export function CreatePublicDashboard({ model }: SceneComponentProps<SharePublicDashboardTab>) {
+  const { isSaveLoading: isLoading, dashboardRef } = model.useState();
+
+  const dashboard = dashboardRef.resolve();
+  const { value: unsupportedDataSources } = useAsync(async () => {
+    const types = getPanelTypes(dashboard);
+    return getUnsupportedDashboardDatasources(types);
+  }, []);
+
+  const hasTemplateVariables = (dashboard.state.$variables?.state.variables.length ?? 0) > 0;
+
+  return (
+    <CreatePublicDashboardComponent
+      onCreate={model.onCreate}
+      isLoading={isLoading}
+      unsupportedDatasources={unsupportedDataSources}
+      unsupportedTemplateVariables={hasTemplateVariables}
+    />
+  );
+}
+
+function getPanelTypes(scene: DashboardScene): string[] {
+  const types = new Set<string>();
+
+  const body = scene.state.body;
+  if (!(body instanceof SceneGridLayout)) {
+    return [];
+  }
+
+  for (const child of body.state.children) {
+    if (child instanceof SceneGridItem) {
+      const ts = panelTypes(child);
+      for (const t of ts) {
+        types.add(t);
+      }
+    }
+
+    if (child instanceof SceneGridRow) {
+      const ts = rowTypes(child);
+      for (const t of ts) {
+        types.add(t);
+      }
+    }
+  }
+
+  return Array.from(types).sort();
+}
+
+export function panelTypes(gridItem: SceneGridItemLike) {
```

Comment:
Nit: this does not return panel types, but data source types. The name is confusing :)